### PR TITLE
Rebase on message-transport

### PR DESCRIPTION
**Pull request by glyn**

### CODE REVIEW

File: glide.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+package: github.com/projectriff/function-sidecar
+import:
+- package: github.com/Shopify/sarama
+  version: ~1.15.0
+- package: github.com/bsm/sarama-cluster
+  version: ~2.1.12
+- package: github.com/giantswarm/retry-go
+  version: ~0.5.0
+- package: github.com/golang/protobuf
+  version: ~1.0.0
```
Comment (ericbottard) [inline glide.yaml]: Is this really a direct dependency, or only though gRPC? If so, should we let grpc drive the version? (assuming it uses a dependency manager that glide understands)

File: glide.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  subpackages:
+  - proto
+- package: github.com/projectriff/message-transport
+  subpackages:
```
Comment (ericbottard) [inline glide.yaml]: Should make `version: master` explicit here, for consistency with other projects

File: cmd/function-sidecar.go [hunk 3]
```diff
@@ -115,69 +112,12 @@ func main() {
...
-						log.Printf("=== Not sending function return value as function did not provide an output channel. Raw result = %s\n", resultMsg)
-					}
-				} else {
-					log.Print("Exiting Kafka Producer loop")
-					return
-				}
-			}
-		}
-	}()
+	carrier.Run(consumer, producer, dispatcher, output)
```
Comment (ericbottard) [inline cmd/function-sidecar.go]: Do you think you can refactor the test in `function-sidecar_test.go` to use this and collaborators?
I really don't like the fact that it runs a compiled binary (that must exist) -- this will hinder the switch to the new Docker scheme, for one thing.

Or we could do this as a separate PR, of course

Comment (glyn) [reply cmd/function-sidecar.go]: @ericbottard I think that's worth trying, but in a separate PR. I raised [issue 45](https://github.com/projectriff/function-sidecar/issues/45) to cover this.

**Comment by glyn** [comment]
@ericbottard I've force pushed a new glide dependencies commit with the changes we agreed.

**Comment by ericbottard** [comment]
Merged as c0893eb
